src/styles/main.scss

```diff
@@ -0,0 +1,82 @@
+body {
+  margin: 0;
+  height: 100vh;
+  display: flex;
+
+  justify-content: center;
+  align-items: center;
+}
+
+// Переменные
+$grey: #eee;
+$purple: #ffbfcb;
+$font-size: 30px;
+$day-size: 100px;
+$border-width: 1px;
+$gap: 1px;
+$paddings: 10px;
+$hover-transition: 0.5s;
+$dayInWe: 7;
+
+// БЭМ стили
+.calendar {
+  width: 726px;
+  display: flex;
+
+  padding: $paddings;
+  flex-wrap: wrap;
+  gap: $gap;
+
+  box-sizing: border-box;
+}
+
+.calendar__day {
+  width: $day-size;
+  height: $day-size;
+
+  background-color: $grey;
+  border: $border-width solid black;
+
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  transition: transform $hover-transition;
+  box-sizing: border-box;
+}
+
+.calendar__day :hover {
```

Comment:
The selector `.calendar__day :hover` has a space before `:hover`, which means it will target any child element of `.calendar__day` that is being hovered over, rather than the `.calendar__day` itself. If the intention is to apply the hover effect to the `.calendar__day` elements, remove the space to make it `.calendar__day:hover`.